intake_bluesky/in_memory.py

```diff
@@ -48,22 +48,19 @@ def __init__(self, handler_registry=None, query=None, **kwargs):
         self._uid_to_run_start_doc = {}
         super().__init__(**kwargs)
 
-    def upsert(self, gen_func, gen_args, gen_kwargs):
-        gen = gen_func(*gen_args, **gen_kwargs)
-        name, run_start_doc = next(gen)
+    def upsert(self, gen_func, start_doc, stop_doc, gen_args, gen_kwargs):
```

Comment:
I think I would find this signature easier to remember if the `gen_func` were next to its args and kwargs. No opion on whether the`*_doc` parameters go first or last, but maybe they shouldn't be in the middle between the func and the args.

This is not a strongly held opinion, feel free to ignore it.